common/math/osqp_interface/CMakeLists.txt

```diff
@@ -12,28 +12,28 @@ endif()
 find_package(ament_cmake_auto REQUIRED)
 ament_auto_find_build_dependencies()
 
+find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
 
 find_package(osqp_vendor REQUIRED)
 
 find_package(osqp REQUIRED)
 get_target_property(OSQP_INCLUDE_DIR osqp::osqpstatic INTERFACE_INCLUDE_DIRECTORIES)
 
-set(EXTERNAL_INCLUDE_DIRS
-  "${EIGEN3_INCLUDE_DIR}"
-  "${OSQP_INCLUDE_DIR}"
-  )
-
-ament_auto_add_library(osqp_interface
+ament_auto_add_library(osqp_interface STATIC
   src/osqp_interface.cpp
   src/csc_matrix_conv.cpp
   include/osqp_interface/osqp_interface.h
   include/osqp_interface/csc_matrix_conv.h
   )
-target_link_libraries(osqp_interface osqp::osqpstatic)
 
-target_include_directories(osqp_interface PUBLIC "${EXTERNAL_INCLUDE_DIRS}")
-# needed so clients of this package don't need to worry about includes of this package
-ament_export_include_directories("${EXTERNAL_INCLUDE_DIRS}")
+target_include_directories(osqp_interface PUBLIC "${OSQP_INCLUDE_DIR}")
+ament_target_dependencies(osqp_interface Eigen3)
+ament_target_dependencies(osqp_interface osqp_vendor)
+
+# crucial so downstream package builds because osqp_interface exposes osqp.h
+ament_export_include_directories("${OSQP_INCLUDE_DIR}")
```

Comment:
Doesn't ament_export_dependencies work too?

Reply:
No, I tried and checked and understand why it fails. First of all, `ament_export_dependencies(osqp_interface)` is called by `ament_auto_package()` below. But it's not enough because it only works with ament-style packages and `osqp` isn't such a package and `osqp_vendor` doesn't expose the necessary info from `osqp`.

An alternative that I just learned today would be to use `ament_export_targets(osqp_interface)`, so I would only have to add the include dir once but hat would mean getting rid of `ament_auto_package()` as far as I understand now. Will check if I get an answer to that question in https://github.com/ament/ament_cmake/issues/292#issuecomment-717967039

Reply:
Played around a bit but more for my own curiousity. Moving to exporting targets would be another step, and from my point of view this PR is ready to be merged @mitsudome-r